### ISSUE TITLE
Error while instrumenting kafka-node

### ISSUE THREAD

**Issue by atd-schubert** [instana/nodejs-sensor#158]
I found an issue with the instana collector and sinek, the highlevel Kafka Client for nodejs.

Sinek depends on node-rdkafka and kafka-node and especially kafka-node causes an error in combination with the instana package.

```
/.../node_modules/@instana/core/src/tracing/instrumentation/messaging/kafka.js:25
  shimmer.wrap(kafka.HighLevelConsumer.prototype, 'emit', shimEmit);
                                       ^

TypeError: Cannot read property 'prototype' of undefined
    at instrument (/.../node_modules/@instana/core/src/tracing/instrumentation/messaging/kafka.js:25:40)
    at Function.patchedModuleLoad [as _load] (/.../node_modules/@instana/core/src/util/requireHook.js:58:36)
    at Module.require (internal/modules/cjs/loader.js:690:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/.../test.js:3:15)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:829:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:622:3)
```

After a short research I found out that node-kafka does not provide a `HighLevelConsumer`, but a `HighLevelProducer`. I also can't find any hint in their repository that there ever was a `HighLevelConsumer`.

I guess that this is some kind of typo. The `HighLevelProducer` has an `emit` and `send` method, so I would guess that one can fix this bug with the following change in `@instana/core/src/tracing/instrumentation/messaging/kafka.js`:

```js
function instrument(kafka) {
  shimmer.wrap(Object.getPrototypeOf(kafka.Producer.prototype), 'send', shimSend);
  shimmer.wrap(kafka.Consumer.prototype, 'emit', shimEmit);
  shimmer.wrap(kafka.HighLevelProducer.prototype, 'emit', shimEmit);
  shimmer.wrap(kafka.HighLevelProducer.prototype, 'send', shimSend);
}
```

I would make a pull-request, but I don't know how your instrumentation and the kafka-node module works. Maybe you can take a look at it and proof and fix it for me?

**Comment by atd-schubert** [comment]
FYI: @basti1302

**Comment by basti1302** [comment]
Hi Arne,

I was on vacation but will look into this in the next few days. Thanks for the report.

**Comment by basti1302** [comment]
> I also can't find any hint in their repository that there ever was a HighLevelConsumer.

There was, but it has been removed with version 4.0.0, see https://github.com/SOHU-Co/kafka-node/blob/master/CHANGELOG.md#breaking-changes.

I have updated the Kafka instrumentation to work with versions >= 4.0.0 as well as older versions. This will be published with the next release.

**Comment by basti1302** [comment]
The fix has landed in release 1.71.0.

**Comment by zauberpony** [comment]
Hey @basti1302, thank you very much, we deployed with the updated package and it seems to work for us now :)
